### PR TITLE
#11 Implement back to top

### PR DESCRIPTION
**Pull request by ruqqq**
Zero dependencies. Pure tailwind and JS.

### CODE REVIEW

File: src/components/back-to-top.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  const scrollToTop = (): void => {
+    window.scrollTo({
+      top: 0,
+      left: 0,
+      behavior: "smooth",
+    });
+  };
+
+  return (
+    <div
```
Comment (relacsua) [inline src/components/back-to-top.tsx:27]: _nitpick_
use `<a> or <button>`  to follow html semantics. I'll let you make the call to edit it or not
Link: https://stackoverflow.com/a/2777280

File: src/components/back-to-top.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  useEffect(() => {
+    window.addEventListener("scroll", listener);
+    return (): void => {
+      window.removeEventListener("scroll", listener);
+    };
+  });
+
+  const scrollToTop = (): void => {
+    window.scrollTo({
```
Comment (relacsua) [inline src/components/back-to-top.tsx]: `scrollTo` not support in safari include iOS phones (are we ok with it) ?
For more info: https://developer.mozilla.org/en-US/docs/Web/API/Window/scrollTo

File: src/components/back-to-top.tsx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      setVisible(false);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener("scroll", listener);
+    return (): void => {
+      window.removeEventListener("scroll", listener);
+    };
+  });
```
Comment (relacsua) [inline src/components/back-to-top.tsx]: not sure how this functions get executed when `scrollToTop` function is called. Mind giving a TLDR?

**Review by relacsua — APPROVED** [review]

File: src/components/back-to-top.tsx (hunk 1 above)
Comment (ruqqq) [reply src/components/back-to-top.tsx:27]: Good point. For accessibility purpose too. Will make the change.

File: src/components/back-to-top.tsx (hunk 2 above)
Comment (ruqqq) [reply src/components/back-to-top.tsx]: Nice catch. Let me test more.

File: src/components/back-to-top.tsx (hunk 3 above)
Comment (ruqqq) [reply src/components/back-to-top.tsx]: The event listener is just to update the React state var `visible`. Not executed in `scrollToTop`.

useEffect here is a substitute of traditional class based React `componentDidMount`.

**Pull request merged**